### PR TITLE
Add missing things

### PR DESCRIPTION
**Pull request by marcinosypka**
* add a missing ifname SetType
* add a missing dynamic SetFlag
* add Elemn counter field
* change Elem fields to be Option

### CODE REVIEW

File: src/schema.rs [hunk 1]
```diff
@@ -249,6 +254,7 @@ pub enum SetFlag {
     Constant,
     Interval,
     Timeout,
+    Dynamic,
```
Comment (jwhb) [inline src/schema.rs:257]: This conforms to the man page as a Sets/Maps flag.

For named Maps this works.
For named Sets it yields an error:

```shell
$ nft -v
nftables v1.0.7 (Old Doc Yak)
$ sudo nft add set ip filter flags_set {type ipv4_addr\; flags constant, interval, dynamic\;}
Error: Could not process rule: Operation not supported
add set ip filter flags_set {type ipv4_addr; flags constant, interval, dynamic;}
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

# it works without dynamic:
$ sudo nft add set ip filter flags_set {type ipv4_addr\; flags constant, interval\;}
$ sudo nft list set ip filter flags_set
table ip filter {
	set flags_set {
		type ipv4_addr
		flags constant,interval
	}
}
```

**Review by jwhb — APPROVED** [review]

File: src/schema.rs (hunk 1 above)
Comment (marcinosypka) [reply src/schema.rs:257]: I can confirm that on nftables v1.0.9 I get the same error.
```
$ nft -v
nftables v1.0.9 (Old Doc Yak #3)

sudo nft add set inet filter flags_set {type ipv4_addr\; flags constant, interval, dynamic\;}
Error: Could not process rule: Operation not supported
add set inet filter flags_set {type ipv4_addr; flags constant, interval, dynamic;}
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Should we somehow enable the dynamic flag only for a named Map and not for a named Set?

Comment (marcinosypka) [reply src/schema.rs:257]: I checked again, it turns out that the `dynamic` flag is mutually exclusive with the `constant` and the `interval` flags.
This command works fine:
```
sudo nft add set inet filter flags_set2 {type ipv4_addr\; flags dynamic\;}
```

Comment (jwhb) [reply src/schema.rs:257]: Great. I'd keep it at that and not model this constraint in this crate. WDYT?

**Pull request merged**